components/dialog/dialog.js

```diff
@@ -94,7 +94,7 @@ class Dialog extends LocalizeStaticMixin(DialogMixin(LitElement)) {
 					</div>
 				</div>
 				<div class="d2l-dialog-content">
-					<div><slot></slot></div>
+					<div style="overflow: auto;"><slot></slot></div>
```

Comment:
Ok I think I figured out what's the issue. Turns out the container for the dialog-content, does not include any margins introduced in the child. According to this: https://stackoverflow.com/questions/27829250/child-margin-doesnt-affect-parent-height,
this is a concept called Collapsing margins. We can fix this by applying overflow: auto;

Reply:
This may be ok - we should make sure to do cross-browser testing on this, as it's still not clear why setting `overflow: auto` vs the default which is `visible` would affect this.  i.e. it's the kinda of things that could easily be different between browsers.

Did you see my comment/question above? Top elements inside of the content slot should not be applying top margin or padding, since the whitespace between the header and content is to be applied by the dialog to ensure consistency. It'd be good to see the DOM that's being used in the content.

Reply:
Yeah i saw it. Here's my code. 
```html
<d2l-dialog title-text="Dismiss Activity">
  <p>Dismissing an activity hides it from Quick Eval, but won't affect the activity</p>
  <p>Dismiss until...</p>
<!-- some other stuff... -->
```
I took some of your feedback and changed those `p` tags into divs. My opinion was that adding margins is something that likely people easily miss, so this is a more "permanent" solution other people don't run into this too.

Reply:
Interesting. I think `<p>` tags are _semantically_ what you want there. You could just do something like:

```
<p style="margin-top:0;">Blah blah</p>
<p style="margin-bottom:0;>Blah blah blah</p>
```

It's not amazing that folks need to know to do this, but the dialog magically doing something also isn't great. There could be totally valid cases for elements to have margins.

Reply:
actually now that I think about it, doing this `style="overflow: auto;"` isn't exactly the best markup.

Plus as you said, cross-browser support could be a pain in the ass. Might just be easier to yell at people if they add margins in the dialog 😆 .

Reply:
oh yeah @dlockhart isn't it best practice for me to not have to worry about implementation details of the dialog?

Reply:
I don't think that's an implementation detail of dialogs... dialogs provide a margin (just like the `<body>` element in documents) and it's up to those using dialogs to ensure that whatever they put inside the dialog doesn't also add its own margins (unless that's what they want).

Apologies if I'm misunderstanding this issue though -- your screenshots in the original issue aren't displaying for me so I don't really know what you're experiencing.

Reply:
dm'ed you the screenshots

Reply:
The `p` elements have `1em` (natively) and `1rem` from `d2l-typography`.  In the context of dialogs however, we would not want first elements to have such margin, and in such case I think it's fair to expect consumers to remove it (I am open to ideas on how to make things easier for them though, if we think this should be collapsing automatically).  I considered using a slotted selector to remove it, however this solution is not great because the first `p` is not necessarily a direct child of the slot.

Default margins on `p` (not what we want):
![image](https://user-images.githubusercontent.com/9042472/69878621-71629500-1293-11ea-9305-2b8dd5684e21.png)

Overriding margins on `p` (what we want):
![image](https://user-images.githubusercontent.com/9042472/69878644-80e1de00-1293-11ea-88f3-c66af9a8d8c5.png)

GitHub seems to be having image issues, but if you click them, you can see them.

Reply:
Just a quick followup - Dave L. and I talked about this offline.  There's another sol'n I think we're going to try to address the height.  Separately, consumers of the dialog should typically remove margin from top elements in dialog content when necessary.

Reply:
cool, ill happily close this PR